app/controllers/users_controller.rb

```diff
@@ -0,0 +1,68 @@
+class UsersController < ApplicationController
+
+  before_action :require_login, :except => [:new, :create]
+  before_action :user_is_current_user?, only: [:edit, :update]
+  def new
+    if current_user
+      redirect_to user_posts_path(current_user.id)
+    else
+      @user = User.new
+    end
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      sign_in(@user)
+      flash[:success] = "Congrats on making your account!"
+
+      redirect_to user_posts_path(current_user.id)
+    else
+      errors = @user.errors.full_messages.join(", ")
+      flash[:sorry] = "We could not create an accout. #{errors}"
+      redirect_to root_path
+    end
+
+  end
+
+  def show
+    @profile = Profile.find_by_user_id(params[:id])
+  end
+
+  def edit
+    @profile = Profile.find_by_user_id(params[:id])
+  end
+
+  def update
+    @profile = Profile.find_by_user_id(params[:id])
```

Comment:
since you grab the profile via the user_id and not by grabbing the user via the auth_token first, i think a user would eventually be able to edit other user's profiles just by manually changing the ID - not sure